addons/purchase_stock/models/purchase.py

```diff
@@ -388,6 +388,16 @@ def _prepare_stock_moves(self, picking):
             res.append(template)
         return res
 
+    @api.model
+    def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, company_id, values, po):
+        res = super()._prepare_purchase_order_line(product_id, product_qty, product_uom, company_id, values, po)
```

Comment:
Do a separate function _prepare_purchase_order_line_from_procurement that call _prepare_purchase_order_line